packages/openactive-integration-tests/test/helpers/flow-stages/opportunity-feed-update.js

```diff
@@ -1,162 +1,319 @@
+   *
+   * Each is an RPDE item.
+   */
+  const opportunityFeedExtractResponses = await Promise.all(testInterfaceOpportunities.map(async (testInterfaceOpportunity, i) => {
+    // Only attempt getMatch if test interface response was successful
+    if (isResponse20x(testInterfaceOpportunity) && testInterfaceOpportunity.body['@id']) {
+      const opportunityId = testInterfaceOpportunity.body['@id'];
+      // If a match for this @id is already being requested, just reuse the same response
+      if (reusableMatchPromises.has(opportunityId)) {
+        return await reusableMatchPromises.get(opportunityId);
+      }
+
+      const matchPromise = requestHelper.getMatch(opportunityId, i, useCacheIfAvailable);
+      reusableMatchPromises.set(opportunityId, matchPromise);
+      return await matchPromise;
+    }
+    return null;
+  }));
+  return opportunityFeedExtractResponses;
+}
 
 /**
- * @typedef {{
- *   position: number;
- *   orderedItem: {
- *     '@type': string,
- *     '@id': string,
- *     [k: string]: any,
- *   };
- *   acceptedOffer: {
- *     '@id': string,
- *     [k: string]: any,
- *   };
- *   'test:primary': boolean;
- *   'test:control': boolean;
- * }} OrderItem
+ * @param {object} args
+ * @param {Promise<ChakramResponse[]>} args.getOpportunitiesFromOpportunityFeedPromise
+ * @returns {Promise<CollectorOutput>}
  */
+async function runOpportunityFeedUpdateCollector({ getOpportunitiesFromOpportunityFeedPromise }) {
+  const opportunityFeedExtractResponses = await getOpportunitiesFromOpportunityFeedPromise;
+  return { opportunityFeedExtractResponses };
+}
 
 /**
- * @param {unknown} opportunity
- * @param {string} opportunityCriteria
+ * @param {object} args
+ * @param {OpportunityCriteria[]} args.orderItemCriteriaList
+ * @param {() => {
+ *   opportunityFeedExtractResponses: ChakramResponse[],
+ * }} args.getterFn Function which gets the state needed for this function.
+ *
+ *   opportunityFeedExtractResponses is supposed to have length equal to the
+ *   length of orderItemCriteriaList.
  */
-function getRandomRelevantOffer(opportunity, opportunityCriteria) {
-  const relevantOffers = getRelevantOffers(opportunityCriteria, opportunity, { harvestStartTime: HARVEST_START_TIME });
-  if (relevantOffers.length === 0) { return null; }
+function itSuccessChecksOpportunityFeedUpdateCollector({ orderItemCriteriaList, getterFn }) {
+  orderItemCriteriaList.forEach((x, i) => {
+    it(`should return 200 on success for request relevant to OrderItem ${i}`, () => {
+      const opportunityFeedExtractResponse = getterFn().opportunityFeedExtractResponses[i];
+      if (!opportunityFeedExtractResponse) throw Error('Pre-requisite step failed: test interface request failed');
 
-  return relevantOffers[Math.floor(Math.random() * relevantOffers.length)];
+      chakram.expect(opportunityFeedExtractResponse).to.have.status(200);
+    });
+  });
```

Comment:
Similarly, this was just moved from `OpportunityFeedUpdateFlowStage.itSuccessChecks`